### PR TITLE
[Rule Tuning] Tuning AWS Rules for SAML Provider Updates and Assumed Roles via STS

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*:
* https://github.com/elastic/ia-trade-team/issues/272

## Summary - What I changed
Per conversation with @aarju, the following AWS OOTB rules had room for improvement.

* `AWS Security Token Service (STS) AssumeRole Usage` -> `AWS STS Temporary Credentials via AssumeRole`
* `AWS SAML Activity` -> `AWS IAM SAML Provider Updated`

For `AWS Security Token Service (STS) AssumeRole Usage`, we changed this rule from a custom KQL query to New Terms. The new terms values will focus solely on `user.id` and `aws.cloudtrail.flattened.request_parameters.roleArn` for uniqueness. It was pointed out that false-positives may occur for environments with third-party IdPs. This remains true for automation workflows as well. New terms should ignore common benign temporary creds retrieved for specific users and roles that are commonly assumed. Additionally, rather than separate `AssumeRole` and `AssumeRoleWithSAML`, we added `*` to account for any assumed role activity. Based on global alert telemetry, we have assumed role activity from various sources (SAML via IdP, web identity, pod identity, etc.) With `user.id: *` added to the query, this field must exist but only exists for assumed role activity where an IAM user is involved, thus ignoring some automated workflows.

For `AWS SAML Activity`, this rule originally seemed to capture multiple SAML authentication workflow activity that may or may not have been related to each other, based on the API calls reported in `event.action`. As a result, `AssumeRoleWithSAML` will be covered in `AWS STS Temporary Credentials via AssumeRole`. This rule now only detects when the IAM SAML provider has had metadata modification. We also increased the risk score associated with this as the result of a true-positive could have higher consequences. Note that this update reduces alerts to ~82 within the last 4 months globally.

Additional Information:
- Updated authors
- Updated false-positive notes for each rule
- Re-formatted queries
- Added `aws.cloudtrail.flattened.request_parameters.roleArn` to `non-ecs.json` as it does not exist in AWS schema we build locally
- Renamed rules to align more with detection logic
- Renamed files accordingly

### AWS IAM SAML Provider Updated
Screenshot 2024-07-16 at 10 11 20 AM

### AWS STS Temporary Credentials via AssumeRole
This is more difficult to test as it is new terms. Instead, knowing that the user ID is used for New Terms regarding AWS IAM and STS events, we can confidently assume lower alert volume or false-positives. There are ~502 unique user.ids for global telemetry from ~8000 alerts in the last 120 days, therefore, this should reduce these.

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `Rule: New`, `Rule: Deprecation`, `Rule: Promote`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: Tuning - Guidelines

These guidelines serve as a reminder set of considerations when tuning an existing rule.

### Documentation and Context

- [ ] Detailed description of the suggested changes.
- [ ] Provide example JSON data or screenshots.
- [ ] Provide evidence of reducing benign events mistakenly identified as threats (False Positives).
- [ ] Provide evidence of enhancing detection of true threats that were previously missed (False Negatives).
- [ ] Provide evidence of optimizing resource consumption and execution time of detection rules (Performance).
- [ ] Provide evidence of specific environment factors influencing customized rule tuning (Contextual Tuning).
- [ ] Provide evidence of improvements made by modifying sensitivity by changing alert triggering thresholds (Threshold Adjustments).
- [ ] Provide evidence of refining rules to better detect deviations from typical behavior (Behavioral Tuning).
- [ ] Provide evidence of improvements of adjusting rules based on time-based patterns (Temporal Tuning).
- [ ] Provide reasoning of adjusting priority or severity levels of alerts (Severity Tuning).
- [ ] Provide evidence of improving quality integrity of our data used by detection rules (Data Quality).
- [ ] Ensure the tuning includes necessary updates to the release documentation and versioning.

### Rule Metadata Checks

- [ ] `updated_date` matches the date of tuning PR merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive. Review to ensure the original intent of the rule is maintained.

### Testing and Validation

- [ ] Validate that the tuned rule's performance is satisfactory and does not negatively impact the stack.
- [ ] Ensure that the tuned rule has a low false positive rate.

**Comment by terrancedejesus** [comment]
This seems to be a bug, making this a blocker.

```
FAILED tests/test_specific_rules.py::TestNewTerms::test_new_terms_fields - AssertionError: aws.cloudtrail.flattened.request_parameters.roleArn not found in ECS, Beats, or non-ecs schemas
```

File: tests/test_specific_rules.py [hunk 1]
```diff
@@ -123,6 +124,8 @@ def test_new_terms_fields(self):
                 # checks if new terms field(s) are in ecs, beats non-ecs or integration schemas
                 queryvalidator = QueryValidator(rule.contents.data.query)
                 _, _, schema = queryvalidator.get_beats_schema([], beats_version, ecs_version)
+                for index_name in rule.contents.data.index:
+                    schema.update(**ecs.flatten(ecs.get_index_schema(index_name)))
```
Comment (terrancedejesus) [inline tests/test_specific_rules.py:128]: non-ecs schema was not being added originally, thus the specific AWS field was not found to be valid.

File: rules/integrations/aws/privilege_escalation_sts_temp_creds_via_assume_role.toml [hunk 2]
```diff
@@ -2,19 +2,28 @@
 creation_date = "2021/05/17"
 integration = ["aws"]
 maturity = "production"
+min_stack_comments = "AWS integration breaking changes, bumping version to ^2.0.0"
```
Comment (terrancedejesus) [inline rules/integrations/aws/privilege_escalation_sts_temp_creds_via_assume_role.toml:5]:
```suggestion
```

**Review by imays11 — APPROVED** [review]

File: rules/integrations/aws/defense_evasion_escalation_aws_suspicious_saml_activity.toml [hunk 3]
```diff
@@ -1,77 +0,0 @@
-[metadata]
-creation_date = "2021/09/22"
-integration = ["aws"]
-maturity = "production"
-updated_date = "2024/05/21"
```
Comment (shashank-elastic) [inline rules/integrations/aws/defense_evasion_escalation_aws_suspicious_saml_activity.toml:5]: Should this not be deprecated like all other rules?

Comment (w0rk3r) [reply rules/integrations/aws/defense_evasion_escalation_aws_suspicious_saml_activity.toml:5]: It was just renamed, the UUID is maintained in the one below

**Review by w0rk3r — APPROVED** [review]
🧙🏼‍♂️

File: rules/integrations/aws/defense_evasion_escalation_aws_suspicious_saml_activity.toml (hunk 3 above)
Comment (shashank-elastic) [reply rules/integrations/aws/defense_evasion_escalation_aws_suspicious_saml_activity.toml:5]: Ah.. ok

**Review by Mikaayenson — APPROVED** [review]

**Pull request merged**